File: src/APIBundle/Entity/File.php
<?php

namespace APIBundle\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity()
 * @ORM\Table(name="files")
 */
class File
{
    /**
     * @ORM\Id
     * @ORM\Column(type="string")
     * @ORM\GeneratedValue(strategy="UUID")
     */
    protected $file_id;

    /**
     * @ORM\Column(type="string")
     */
    protected $file_name;

    /**
     * @ORM\Column(type="string")
     */
    protected $file_path;

    /**
     * @ORM\Column(type="string")
     */
    protected $file_type;

    /**
     * @ORM\OneToMany(targetEntity="FileTag", mappedBy="tag_file")
     * @var FileTag[]
     */
    protected $file_tags;

    protected $file_url;

    public function __construct()
    {
        $this->file_tags = new ArrayCollection();
    }

    public function getFileId()
    {
        return $this->file_id;
    }

    public function setFileId($id)
    {
        $this->file_id = $id;
        return $this;
    }

    public function getFileName()
    {
        return $this->file_name;
    }

    public function setFileName($name)
    {
        $this->file_name = $name;
        return $this;
    }

    public function getFilePath()
    {
        return $this->file_path;
    }

    public function setFilePath($path)
    {
        $this->file_path = $path;
        return $this;
    }

    public function getFileType()
    {
        return $this->file_type;
    }

    public function setFileType($type)
    {
        $this->file_type = $type;
        return $this;
    }

    public function getFileUrl()
    {
        return $this->file_url;
    }

    public function setFileUrl($url)
    {
        $this->file_url = $url;
        return $this;
    }

    public function getFileTags()
    {
        return $this->file_tags;
    }

    public function setFileTags($tags)
    {
        $this->file_tags = $tags;
        return $this;
    }

    public function buildUrl($protocol, $hostname)
    {
        $this->file_url = $protocol . '://' . $hostname . '/upload/' . $this->file_path;
    }
}

File: openex-player/src/main/java/io/openex/database/model/File.java
package io.openex.database.model;

import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;

@Entity
@Table(name = "files")
public class File implements Base {
    @Id
    @Column(name = "file_id")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    private String id;

    @Column(name = "file_name")
    private String name;

    @Column(name = "file_path")
    private String path;

    @Column(name = "file_type")
    private String type;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }
}
